### PR TITLE
S3FS

### PR DESCRIPTION
**Pull request by jbickar**
@sherakama likewise with #38, this should not affect existing installs; this just prepares us to be able to use S3FS. Depending on order of operations, this may or may not merge cleanly with the simplesamlphp branch.

Please do not delete this branch after merging.

(@mistermarco FYI)

### CODE REVIEW

**Comment by jbickar** [comment]
AWSDISC-140

**Comment by jbickar** [comment]
Merge conflict is in the .make file. I'll push up a fix.

**Comment by jbickar** [comment]
Why do you hate me, git?

**Comment by jbickar** [comment]
OK, I don't know why github is saying this can't be automatically merged, but I do want s3fs in the .make file. Raar.

File: make/stanford.make [hunk 1]
```diff
@@ -250,4 +268,4 @@ projects[stanford][type] = "profile"
 projects[stanford][download][type] = "git"
 projects[stanford][download][url] = "git@github.com:SU-SWS/Stanford-Drupal-Profile.git"
 ; projects[stanford][download][tag] = 7.x-1.x-20140907
-projects[stanford][download][branch] = 7.x-1.x-dev
\ No newline at end of file
+projects[stanford][download][branch] = 7.x-1.x-s3fs
```
Comment (sherakama) [inline make/stanford.make]: I think this should be re-named to 7.x-2.x

This is an environment specific branch and we want to slowly get rid of this type of work. This is also the next version of this profile with a significant enough change it warrants the 2.x name.

File: stanford.profile [hunk 2]
```diff
@@ -83,32 +83,47 @@ function stanford_sites_tasks() {
   /**
    * File system settings.
    */
+  $enable_s3fs = variable_get('enable_s3fs', 0);
+  if($enable_s3fs == 1) {
```
Comment (sherakama) [inline stanford.profile]: space between "if" & "("

File: stanford.profile [hunk 3]
```diff
@@ -83,32 +83,47 @@ function stanford_sites_tasks() {
...
-  //check that the public directory exists; create it if it does not
-  system_check_directory($element);
+    // run drush s3fs-refresh-cache (or equivalent function)?
+    // From s3fs_update_7000().
+    //$config = _s3fs_get_config();
+    //if (!empty($config['bucket']) && !empty($config['region'])) {
+    //  _s3fs_refresh_cache($config);
+    //}
+  }
+  else{
```
Comment (sherakama) [inline stanford.profile]: Space between else and opening bracket.

File: stanford.profile [hunk 4]
```diff
@@ -83,32 +83,47 @@ function stanford_sites_tasks() {
...
+    //Set private directory
+    $private_directory = 'sites/default/files/private';
+    variable_set('file_private_path', $private_directory);
+    //system_check_directory() is expecting a $form_element array
```
Comment (sherakama) [inline stanford.profile]: Remove this comment.

File: stanford.profile [hunk 5]
```diff
@@ -83,32 +83,47 @@ function stanford_sites_tasks() {
...
+    $element = array();
+    $element['#value'] = $private_directory;
+    //check that the public directory exists; create it if it does not
```
Comment (sherakama) [inline stanford.profile]: Comments need a space between comment start // 
Comments need to start with a capital letter and close with a full stop.

File: stanford.profile [hunk 6]
```diff
@@ -83,32 +83,47 @@ function stanford_sites_tasks() {
...
-  variable_set('file_public_path', $public_directory);
-  //Set default scheme to public file handling
-  variable_set('file_default_scheme', 'public');
-  //system_check_directory() is expecting a $form_element array
-  $element = array();
-  $element['#value'] = $public_directory;
-  $element['#name'] = 'file_public_path';
-  //check that the public directory exists; create it if it does not
-  system_check_directory($element);
+    //Set public directory
```
Comment (sherakama) [inline stanford.profile]: Add space and full stop.

File: stanford.profile [hunk 7]
```diff
@@ -83,32 +83,47 @@ function stanford_sites_tasks() {
...
+    $public_directory = 'sites/default/files';
+    variable_set('file_public_path', $public_directory);
+    //Set default scheme to public file handling
+    variable_set('file_default_scheme', 'public');
+    //system_check_directory() is expecting a $form_element array
```
Comment (sherakama) [inline stanford.profile]: Comment line not needed.

File: stanford.profile [hunk 8]
```diff
@@ -83,32 +83,47 @@ function stanford_sites_tasks() {
...
+    $element = array();
+    $element['#value'] = $public_directory;
+    $element['#name'] = 'file_public_path';
+    //check that the public directory exists; create it if it does not
+    system_check_directory($element);
+  }
 
   //Enable the stanford_sites_helper module and the webauth module
```
Comment (sherakama) [inline stanford.profile]: Add space and full stop.

File: stanford.profile [hunk 9]
```diff
@@ -83,32 +83,47 @@ function stanford_sites_tasks() {
...
   //Do this now rather than in .info file because it's looking for the administrator role and errors out otherwise
   module_enable(array('stanford_sites_helper', 'webauth'));
-  
+
   //Make the Seven admin theme use our favicon
```
Comment (sherakama) [inline stanford.profile]: Add space and full stop.

File: stanford.profile [hunk 10]
```diff
@@ -259,9 +274,9 @@ function stanford_sites_add_webauth_user($sunet, $name = '', $email = '') {
     // hide Local Drupal user login block. User 1 can still login from /user
     variable_set(webauth_allow_local, 0);
 
-    watchdog('Stanford Profile','Created user: %user', array('%user' => $name));    
+    watchdog('Stanford Profile','Created user: %user', array('%user' => $name));
```
Comment (sherakama) [inline stanford.profile]: Missing space between function params.

File: stanford.profile [hunk 11]
```diff
@@ -259,9 +274,9 @@ function stanford_sites_add_webauth_user($sunet, $name = '', $email = '') {
...
   }
   else {
-    watchdog('Stanford Profile','Could not create duplicate user: %user', array('%user' => $name)); 
+    watchdog('Stanford Profile','Could not create duplicate user: %user', array('%user' => $name));
```
Comment (sherakama) [inline stanford.profile]: Missing space between function params.

File: stanford.profile (hunk 2 above)
Comment (jbickar) [inline stanford.profile]: 50% of the time, this gets me every time.

File: stanford.profile (hunk 10 above)
Comment (jbickar) [inline stanford.profile]: I blame @mistermarco for that

File: stanford.profile (hunk 4 above)
Comment (jbickar) [inline stanford.profile]: ¿porque?

File: make/stanford.make (hunk 1 above)
Comment (jbickar) [inline make/stanford.make]: I disagree on this (renaming to 2.x), although not particularly strongly. We are adding functionality, and not removing any functionality. These code updates are backwards-compatible with earlier installs, and should not break anything existing. (I did change it from 7.x-1.x-s3fs to 7.x-1.x-dev)

Comment (jbickar) [inline make/stanford.make]: disagree

**Comment by sherakama** [comment]
Ive created a new branch for this work. You can carry on development of environment related tasks in the 2.x branch.

**Pull request closed without merging**